### ISSUE TITLE
Issues with CoreData

### ISSUE THREAD

**Issue by deivitaka** [Iterable/swift-sdk#554]
- Reproducible in the sample app
- v6.4.2

Using argument `-com.apple.CoreData.ConcurrencyDebug 1` exceptions are thrown, and we are also having crash logs regarding misuse of the context (called from the wrong thread?)

Specifically in `CoreDataUtil::count(context:, entity:)`

Crash 1: 
```
Crashed: NSManagedObjectContext 0x283722a40
0  libobjc.A.dylib                0x3fc8 object_getClass + 48
1  CoreData                       0x5aa2c -[NSManagedObject _validateForSave:] + 196
2  CoreData                       0x9fbd4 -[NSManagedObjectContext _validateObjects:forOperation:error:exhaustive:forSave:] + 860
3  CoreData                       0x4a5a0 -[NSManagedObjectContext _prepareForPushChanges:] + 248
4  CoreData                       0x2da8c -[NSManagedObjectContext save:] + 544
5  myapp                   0x2fa8f4 specialized IterableTaskScheduler.schedule(apiCallRequest:context:scheduledAt:) + 168 (IterableCoreDataPersistence.swift:168)
6  myapp                   0x30cde0 specialized OfflineRequestProcessor.sendIterableRequest(requestGenerator:successHandler:failureHandler:identifier:) + 265 (<compiler-generated>:265)
7  myapp                   0x3072c4 protocol witness for RequestProcessorProtocol.track(event:dataFields:onSuccess:onFailure:) in conformance OfflineRequestProcessor + 4303303364 (<compiler-generated>:4303303364)
8  myapp                   0x322f08 specialized static IterableAPI.track(event:dataFields:) + 132 (RequestHandler.swift:132)
```

Crash 2:
```
Crashed: NSManagedObjectContext 0x2819027d0
0  libswiftCore.dylib             0x3dcd08 swift_isUniquelyReferenced_nonNull_native + 38
1  myapp                   0x306100 OfflineRequestProcessor.NotificationListener.futureFromTask(withTaskId:) + 25 (Promise.swift:25)
2  myapp                   0x30ce10 specialized OfflineRequestProcessor.sendIterableRequest(requestGenerator:successHandler:failureHandler:identifier:) + 267 (<compiler-generated>:267)
3  myapp                   0x3072c4 protocol witness for RequestProcessorProtocol.track(event:dataFields:onSuccess:onFailure:) in conformance OfflineRequestProcessor + 4367102660 (<compiler-generated>:4367102660)
4  myapp                   0x322f08 specialized static IterableAPI.track(event:dataFields:) + 132 (RequestHandler.swift:132)
```

**Comment by tapashmajumder** [comment]
Thanks for brining this to our attention. We will be looking into this urgently. 
To be clear, this only happens if you enable core data concurrency debug, right?

Also, how did you turn offline mode on for your project? This feature is not enabled by default.

**Comment by tapashmajumder** [comment]
This is very helpful information. We are looking at this urgently. In the meantime please let us know if you would want us to turn off online mode for you.

**Comment by tapashmajumder** [comment]
Hi @deivitaka  We have a PR to resolve this issue. Would you be able to test it?

https://github.com/Iterable/swift-sdk/pull/555

**Comment by deivitaka** [comment]
after https://github.com/Iterable/swift-sdk/pull/555/commits/a3e0e02632637afa0ac921a485a7c4e24b2ec373 looks good

**Comment by tapashmajumder** [comment]
@deivitaka Can you please try and see if this PR/branch resolves the problem you are having? If so, we will create a new release with the fixes.

**Comment by tapashmajumder** [comment]
I am fixing some unit tests in the meantime.

**Comment by tapashmajumder** [comment]
Thank you very much!

**Comment by roninopf** [comment]
Hi @deivitaka, just closing this out that this is released in 6.4.3. Thanks for finding this and helping with the fix!

If any new issues come up, please feel free to open a new issue or message your CSM.
